File: C++/206.反转链表.cpp
// 206. 反转链表

// 给你单链表的头节点 head ，请你反转链表，并返回反转后的链表。

#include "header.h"

class Solution {
public:
    // 三指针
    ListNode* reverseList(ListNode* head)
    {
        ListNode *pre = nullptr, *node = head, *next = head;
        while (node != nullptr) {
            next = node->next;
            node->next = pre;
            pre = node;
            node = next;
        }
        return pre;
    }
};

File: leetcode/02链表/206. 反转链表.go
package main

// 206. 反转链表
// 给你单链表的头节点 head ，请你反转链表，并返回反转后的链表。

// reverseList .
func reverseList(head *ListNode) *ListNode {
	// 前一个节点、当前节点、下一个节点
	pre, node, next := (*ListNode)(nil), head, head

	for node != nil {
		next = node.Next
		node.Next = pre
		pre = node
		node = next
	}

	return pre
}
